x-pack/plugins/cases/server/client/utils.test.ts

```diff
@@ -24,6 +24,106 @@ import {
 
 describe('utils', () => {
+  describe('buildFilter', () => {
+    it('returns undefined if filters is undefined', () => {
+      expect(buildFilter({ filters: undefined, field: 'abc', operator: 'or' })).toBeUndefined();
+    });
+
+    it('returns undefined if filters is is an empty array', () => {
+      expect(buildFilter({ filters: [], field: 'abc', operator: 'or' })).toBeUndefined();
+    });
+
+    it('returns a KueryNode using or operator', () => {
+      expect(buildFilter({ filters: ['value1'], field: 'abc', operator: 'or' }))
+        .toMatchInlineSnapshot(`
+        Object {
+          "arguments": Array [
+            Object {
+              "isQuoted": false,
+              "type": "literal",
+              "value": "cases.attributes.abc",
+            },
+            Object {
+              "isQuoted": false,
+              "type": "literal",
+              "value": "value1",
+            },
+          ],
+          "function": "is",
+          "type": "function",
+        }
+      `);
+    });
+
+    it("returns multiple nodes or'd together", () => {
+      expect(buildFilter({ filters: ['value1', 'value2'], field: 'abc', operator: 'or' }))
+        .toMatchInlineSnapshot(`
+        Object {
+          "arguments": Array [
+            Object {
+              "arguments": Array [
+                Object {
+                  "isQuoted": false,
+                  "type": "literal",
+                  "value": "cases.attributes.abc",
+                },
+                Object {
+                  "isQuoted": false,
+                  "type": "literal",
+                  "value": "value1",
+                },
+              ],
+              "function": "is",
+              "type": "function",
+            },
+            Object {
+              "arguments": Array [
+                Object {
+                  "isQuoted": false,
+                  "type": "literal",
+                  "value": "cases.attributes.abc",
+                },
+                Object {
+                  "isQuoted": false,
+                  "type": "literal",
+                  "value": "value2",
+                },
+              ],
+              "function": "is",
+              "type": "function",
+            },
+          ],
+          "function": "or",
+          "type": "function",
+        }
+      `);
+    });
+
+    it('does not escape special kql characters in the filter values', () => {
+      const specialCharacters = 'awesome:()\\<>"*';
+
+      expect(buildFilter({ filters: [specialCharacters], field: 'abc', operator: 'or' }))
+        .toMatchInlineSnapshot(`
+        Object {
+          "arguments": Array [
+            Object {
+              "isQuoted": false,
+              "type": "literal",
+              "value": "cases.attributes.abc",
+            },
+            Object {
+              "isQuoted": false,
+              "type": "literal",
+              "value": "awesome:()\\\\<>\\"*",
```

Comment:
It's interesting that some of this is escaped, in the integration tests the slash still seem to work though 🤷‍♂️